server/src/sim.rs

```diff
@@ -420,6 +445,82 @@ impl Sim {
             }
         }
     }
+
+    /// Add the given entity to the given inventory
+    fn add_to_inventory(&mut self, inventory_id: EntityId, entity_id: EntityId) {
+        let mut inventory = self
+            .world
+            .get::<&mut Inventory>(*self.entity_ids.get(&inventory_id).unwrap())
+            .unwrap();
+        inventory.contents.push(entity_id);
+        self.accumulated_changes
+            .inventory_additions
+            .push((inventory_id, entity_id));
+    }
+
+    /// Remove the given entity from the given inventory. Note that this does not destroy the entity.
+    fn remove_from_inventory(&mut self, inventory_id: EntityId, entity_id: EntityId) {
+        let mut inventory = self
+            .world
+            .get::<&mut Inventory>(*self.entity_ids.get(&inventory_id).unwrap())
+            .unwrap();
+        inventory.contents.retain_mut(|e| *e != entity_id);
+        self.accumulated_changes
+            .inventory_removals
+            .push((inventory_id, entity_id));
+    }
+
+    /// Check if the given entity is in the given inventory
+    fn is_in_inventory(&self, inventory_id: EntityId, entity_id: EntityId) -> bool {
```

Comment:
I would have done that had I realized that checking the material type before checking whether the entity was in the inventory was an option. I thought I needed to separate them because of that additional check. Thanks for getting me to take another look at this!